docs/user/explanations/system-simulation-adapters.rst

```diff
@@ -0,0 +1,154 @@
+------------------------
+
+The following system adapter is a simple `CommandAdapter`. This Can be used to query the
+SystemSimulationComponent for a list of ID's of the components it contains; and to
+return the wiring map of the components.
+
+.. code-block:: python
+
+    class SystemSimulationAdapter(BaseSystemSimulationAdapter, CommandAdapter):
+
+        _byte_format: ByteFormat = ByteFormat(b"%b\r\n")
+
+        @RegexCommand(r"ids", False, "utf-8")
+        async def get_component_ids(self) -> bytes:
+            """Returns a list of ids for all the components in the system simulation."""
+            return str(self._components.keys()).encode("utf-8")
+
+        @RegexCommand(r"wiring", False, "utf-8")
+        async def get_wiring(self) -> bytes:
+            """Returns the wiring object used by the nested scheduler."""
+            return str(self._wiring).encode("utf-8")
+
+To use this adapter we would need to put it in an adapter container with TcpIo and
+assign it as an argument in the SystemSimulationComponent.
+
+.. code-block:: python
+
+    @pydantic.v1.dataclasses.dataclass
+    class NestedAmplifierWithAdapter(ComponentConfig):
+        """Simulation of a nested amplifier with a CommandAdapter."""
+
+        name: ComponentID
+        inputs: Dict[PortID, ComponentPort]
+        components: List[ComponentConfig]
+        expose: Dict[PortID, ComponentPort]
+
+        def __call__(self) -> Component:  # noqa: D102
+            return SystemSimulationComponent(
+                name=self.name,
+                components=self.components,
+                expose=self.expose,
+                adapter=AdapterContainer(
+                    SystemSimulationAdapter(),
+                    TcpIo(host="localhost", port=25560),
+                ),
+            )
+
+
+
+This ComponentConfig ``NestedAmplifierWithAdapter`` can be used as any other component.
+
+Below is a very simple simulation with a system simulation component. It is a
+source, which is connected to a `SystemSimulationComponent` which only contains a
+single amplifier. The output of this amplfier is fed to the output of the system
+simulation component, which is wired to a sink. 
+
+
+.. code-block:: yaml
+
+    - type: tickit.devices.source.Source
+    name: source
+    inputs: {}
+    value: 10.0
+
+    - type: examples.adapters.system_simulation_adapter_config.NestedAmplifierWithAdapter
+    name: nested-amp
+    inputs:
+        input_1:
+        component: source
+        port: value
+    components:
+        - type: examples.devices.amplifier.Amplifier
+        name: amp
+        inputs:
+            initial_signal:
+            component: external
+            port: input_1
+        initial_amplification: 2
+    expose:
+        output_1:
+        component: amp
+        port: amplified_signal
+
+    - type: tickit.devices.sink.Sink
+    name: external_sink
+    inputs:
+        sink_1:
+        component: nested-amp
+        port: output_1
```

Comment:
Trying to not be too nitpicky, but since this is docs, would benefit from being autoformatted so that the yaml is valid on copy-paste

Reply:
I don't know what that means

Reply:
```suggestion
- type: tickit.devices.source.Source
  name: source
  inputs: {}
  value: 10.0

- type: examples.adapters.system_simulation_adapter_config.NestedAmplifierWithAdapter
  name: nested-amp
  inputs:
    input_1:
      component: source
      port: value
  components:
    - type: examples.devices.amplifier.Amplifier
      name: amp
      inputs:
        initial_signal:
          component: external
          port: input_1
      initial_amplification: 2
  expose:
    output_1:
      component: amp
      port: amplified_signal

- type: tickit.devices.sink.Sink
  name: external_sink
  inputs:
    sink_1:
      component: nested-amp
      port: output_1

```

Reply:
Ah, I see. I didn't even notice it had shifted, I just pasted it in.